### DISCUSSION TITLE
ROS2 JointStatePublisher in the Browser

### DISCUSSION THREAD

**Discussion by kavikode** [RobotWebTools/roslibjs#751] [Q&A] [answered]
Which repo branch can be used to replicate JointStatePublisher in the Browser for ROS2 Humble?

Link: https://wiki.ros.org/joint_state_publisher_js/Tutorials/Using%20JointStatePublisher%20in%20the%20Browser

I'd like to use the Tiago URDF simulation from https://github.com/pal-robotics/tiago_simulation
 
Also, since we are now using XACRO format for URDF, how do we convert it in ROS2 and successfully visualize and interact with the URDF in the browser?

Please provide the steps necessary. My hope it is possible using ROS 2.

Thank you so much

**Comment by EzraBrooks** [comment, marked as answer]
Hi, as this is a question and not a bug, I've converted it from an issue to a discussion.

My recommendation would be to subscribe to the `/joint_states` topic using roslibjs and then apply those joint states to a URDF from either [`ros3djs`](https://github.com/RobotWebTools/ros3djs) or [`urdf-loaders`](https://github.com/gkjohnson/urdf-loaders).
